Add styling for tables

thaJeztah opened the pull request:
This adds some basic styling for tables in the documentation. Styling is based on a mix of GitHub tables and tweaked a bit to match the documentation theme.

With this change inline-styles of existing tables can be removed and html-tables can be replaced with plain MarkDown tables in most cases.

Some examples;

Before:
![remote-libraries-before](https://cloud.githubusercontent.com/assets/1804568/6472023/add5968a-c1ef-11e4-9c20-ea4cc8ea8eee.png)

After:
![remote-libraries-after](https://cloud.githubusercontent.com/assets/1804568/6472029/b6a8de8e-c1ef-11e4-95e3-9719d5753a1f.png)

Plain MarkDown table:
![inline-table](https://cloud.githubusercontent.com/assets/1804568/6472044/cb9ae706-c1ef-11e4-83f4-bb7169116668.png)

moxiegirl commented:
@thaJeztah is this a....Proposal? :-D  The POMT look pretty clean. Can I still use my html tables?  Sometimes, I need the control to do that.  For example, I want a table without borders for some information.

thaJeztah commented:
@moxiegirl yes, it's a proposal. The problem with the current tables is that everyone will style their tables differently, which will also make it problematic if (at some time) the theme is updated.

Adding (default) styles to the stylesheet, will make the tables look consistent and allow updating them with a new style if desired.

If you specify inline-styles in the table, you will always be able to override styles from the stylesheet, but in general, this should be discouraged.

I'm open to suggestions though, if the default style has to be tweaked some more.  I like that this proposal has some more white space inside the table, and a margin around the table :)

In general, I'm not always fond of borders, but looking at the examples, they *may* be a bit easier to read.

moxiegirl commented:
@thaJeztah There are situations where the content calls for different types of table styling. For example, I use tables without borders to align the output of say `docker ps` -- just easy and more consistently presented with an HTML table w/o borders. 

I think your layout is an improvement.

thaJeztah commented:
@moxiegirl do you have a link to the page you're using the table without borders?

I can add some variations for tables, so that you can add a "class" on the table and make it look different, e.g.

`<table class="plain">`, `<table class="cli-output">` or `<table class="no-borders">`

I prefer a "descriptive" name for the classes; describe the *intent*, not what it *looks like*. This makes it less tied to the theme (for example, naming a class "dark blue" isn't very useful if the theme switches to a "green" color-scheme :))

moxiegirl commented:
@thaJeztah Great idea! Yes, of course a style would be fine. This is work I'm doing for the birthday so it isn't out yet. I'm still drafting it.  Your PR might not make it in before my release date.  So, just propose a class and I'll add it after the fact.

Examples are here on my draft:
http://52.11.52.251:8000/helloworld/set-up-dev-env/

thaJeztah commented:
@moxiegirl alright, since it is for cli-output, I'll add a class `cli-output`, (or do you have a better name in mind?)

I wonder though, if a table for the `docker ps` output is easy to maintain. Most peeps contributing to docker like to just copy/paste the output. A `<pre>` (code-block), is easier for them.

But since this is a separate document, perhaps it's no problem,

Really going off-topic here (without revealing what you're working on); some (possibly useful tips);

First, you have a typo in there; "set yp your local machine"

I see you're trying to crop your screenshots, but there's a black border around some of them. I looks like you're working on OS X. To create a screenshot of a *single window*, press <kbd>CMD</kbd>+<kbd>Shift</kbd>+<kbd>4</kbd>, then press the <kbd>space</kbd>. then hover over the Window you want to take a screenshot to highlight it and *click* to take the screenshot.

To take a screenshot of specific section of your screen, press <kbd>CMD</kbd>+<kbd>Shift</kbd>+<kbd>4</kbd>, you'll see a cross-hair cursor. Then drag around the section you want to take a screenshot of.

By default, OS X will also include the Window's drop-shadow in screenshots. To disable this, look at [this article](http://osxdaily.com/2011/05/23/disable-shadow-screen-shots-mac/). In short; do this on the command-line;

    defaults write com.apple.screencapture disable-shadow -bool true
    killall SystemUIServer

And finally; to optimize your screenshots *losslessly*, you can use this utilty; https://imageoptim.com/.

moxiegirl commented:
@thaJeztah I'm using Snagit for screenshots.  Thanks for the tips though, I'll look at them.

This piece is a _rough draft_ which means I don't know it isn't ready for a formal copy edit. I've only tested against a single OS and expect extensive rewrites.  Thanks, though, for pointing out the typo.

As Voltaire said "the perfect is the enemy of the good" which is too say it would be nice to clean up the black borders etc but I might not get to it with my deadline --- should you still see them in the future.

thaJeztah commented:
> but I might not get to it with my deadline --- should you still see them in the future

No prob! I saw you struggling there, and those shortcuts may well save you some time, so I thought to share them with you :)

thaJeztah commented:
Added a `<table class="code">` variation. I copied your table, removed all classes and added a "code" class to the table-tag. It looks like this;

![code-and-regular](https://cloud.githubusercontent.com/assets/1804568/6474397/abbb44f6-c200-11e4-8655-c949afa1ca99.png)

moxiegirl commented:
@thaJeztah Saweet, I like the slight grey to draw attention...you are one fast coder!

Maybe leave off the bold heading though --- doesn't happen at the console -- least not mine. :-D

thaJeztah commented:
Fixed! (Another option would be to *not* use `<th>` for those)

thaJeztah commented:
@SvenDowideit do you think it's possible to merge this? I'm planning to make some additional PRs for parts in the docs where tables would be a nice improvement. Although that would be possible with inline styles, having it in the CSS would be a better option :smile:

moxiegirl commented:
LGTM

SvenDowideit commented:
yup, LGTM - merci-much :)

thaJeztah commented:
Thanks for the merge!
